experiment/runner.py

```diff
@@ -157,6 +161,56 @@ def _unpack_clusterfuzz_seed_corpus(fuzz_target_path, corpus_directory):
               seed_corpus_archive_path)
 
 
+def sample_corpus(corpus_dir,
+                  random_seed,
+                  dest_dir=None,
+                  distribution='EXP',
+                  mean_seed_util=0.2):
+    """Samples a pseudo-random number of files from the input corpus. By
+    default sampling is done in-place, destructively, removing unsampled
+    files. Will sample (mean_seed_util * number of seeds) files on
+    average, according to the specified distribution. Sampling is
+    deterministic wrt the random seed"""
+
+    gen = random.Random(random_seed)
+    npgen = numpy.random.RandomState(random_seed)
+
+    inplace = dest_dir is None
+
+    corpus_paths = [
+        f for f in glob.glob(f'{corpus_dir}/**/*', recursive=True)
+        if os.path.isfile(f)
+    ]
+    corpus_paths.sort()  # need to be ordered for deterministic sampling
+    logs.info('Sampling from %d files in seed corpus dir %s.',
+              len(corpus_paths), corpus_dir)
+
+    num_seeds = len(corpus_paths)
+
+    if distribution == 'UNIFORM':
+        trial_num_seeds = gen.randint(
+            1, int(numpy.round(mean_seed_util * num_seeds)))  # inclusive []
```

Comment:
Here is a question that I wanted to ask earlier but was unsure if I missed anything:
If the benchmark does not come with a seed corpus, or `mean_seed_util * num_seeds` is less than 0.5 (i.e., evaluates to `randint(1, 0)`), would this cause a `ValueError` because the upper bound (`0`) is smaller than the lower bound (`1`)?

Reply:
Previously I thought we would only expect to run experiments on preselected benchmarks that have enough seeds, but I guess my assumption was wrong?

Reply:
Yeah, this is a good point. I'm not sure restricting the benchmarks makes sense, since people may use their own custom seed corpora and its possible the default corpora might change over time.

Also, does Fuzzbench automatically insert a file for the "empty" corpus? I seem to remember seeing a "hi.txt" file floating around on "empty" runs, maybe because AFL won't run with an empty input folder?

If not can add an assertion that the size of the seed corpus should be non-zero to make things more clear when it fails, but I expect it won't come up very frequently since the feature is opt-in for now.

Reply:
Looking at the results from the run now and it seems look like `lcms_cms_transform_fuzzer` has no coverage to start, so I guess there is no issue